File: unity/rc0_runtime/Assets/RC0Runtime/Core/RuntimeCommand.cs
using System.Collections.Generic;

namespace RC0.Runtime.Core
{
    /// <summary>Flutter → Unity command envelope (v1 protocol).</summary>
    public sealed class RuntimeCommand
    {
        public int v;
        public string sessionId;
        public string module;
        public string action;
        public Dictionary<string, object> payload;

        public RuntimeCommand()
        {
            payload = new Dictionary<string, object>();
        }
    }

    /// <summary>Unity → Flutter event envelope.</summary>
    public sealed class RuntimeEvent
    {
        public int v;
        public string sessionId;
        public string module;
        public string eventName;
        public Dictionary<string, object> payload;

        public RuntimeEvent()
        {
            payload = new Dictionary<string, object>();
        }
    }
}

File: lib/runtime_3d/core/runtime_command.dart
/// JSON command envelope (Flutter → Unity, v1).
class RuntimeCommand {
  const RuntimeCommand({
    required this.sessionId,
    required this.module,
    required this.action,
    this.payload = const {},
    this.version = 1,
  });

  final int version;
  final String sessionId;
  final String module;
  final String action;
  final Map<String, dynamic> payload;

  Map<String, dynamic> toJson() => {
        'v': version,
        'sessionId': sessionId,
        'module': module,
        'action': action,
        'payload': payload,
      };
}
